src/modules/powerrename/ui/PowerRenameUI.cpp

```diff
@@ -791,40 +789,56 @@ void CPowerRenameUI::_MoveControl(_In_ DWORD id, _In_ DWORD repositionFlags, _In
     {
         flags |= SWP_NOSIZE;
     }
+    RECT rc = { 0 };
+    GetWindowRect(m_hwnd, &rc);
+    int mainWindowWidth = rc.right - rc.left;
+    int mainWindowHeight = rc.bottom - rc.top;
 
     RECT rcWindow = { 0 };
     GetWindowRect(hwnd, &rcWindow);
-
-    int cx = RECT_WIDTH(rcWindow);
-    int cy = RECT_HEIGHT(rcWindow);
-
     MapWindowPoints(HWND_DESKTOP, GetParent(hwnd), (LPPOINT)&rcWindow, 2);
-
     int x = rcWindow.left;
     int y = rcWindow.top;
+    int width = rcWindow.right - rcWindow.left;
+    int height = rcWindow.bottom - rcWindow.top;
 
-    if (repositionFlags & Reposition_X)
-    {
-        x += xDelta;
-    }
-
-    if (repositionFlags & Reposition_Y)
-    {
-        y += yDelta;
-    }
+    UINT currentDPI = 0, dummy;
+    DPIAware::GetScreenDPIForWindow(m_hwnd, currentDPI, dummy);
+    float scale = (float)currentDPI / m_initialDPI;
 
-    if (repositionFlags & Reposition_Width)
+    switch (id)
     {
-        cx += xDelta;
-    }
-
-    if (repositionFlags & Reposition_Height)
-    {
-        cy += yDelta;
+    case IDC_EDIT_SEARCHFOR:
+    case IDC_EDIT_REPLACEWITH:
+        width = mainWindowWidth - m_itemsPositioning.searchReplaceWidthDiff * scale;
```

Comment:
let's not produce warnings and add `static_cast<int>` to all multiplications by `scale`